### PR TITLE
Assignment 1: mmoran314 (Michael Moran) and BenFristadSchool (Ben Fristad)

### PR DESCRIPTION
**Pull request by mmoran314**
Michael Moran mmoran314
Ben Fristad BenFristadSchool
Added feature to display correct answer if incorrect answer is entered. Added unit test for new method, GetCorrectAnswer

### CODE REVIEW

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -89,8 +95,17 @@ public static Question[] LoadQuestions(string filePath)
                 question.Answers[1] = answer2;
                 question.Answers[2] = answer3;
                 question.CorrectAnswerIndex = correctAnswerIndex;
+
+                questions[i] = question; // Added in this line of code
             }
             return questions;
         }
+        
+        public static string GetCorrectAnswer(Question question) // Added method that returns correct answer
```
Comment (StrangeRanger) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:104]: Looks like it works as it should. No problems here. I like this idea of an extra feature, good job.

**Comment by Malcolm-Shepherd** [comment]
# Assignment Details

- Pull Request title includes "Assignment 1" in the title ✔
- Pull Request targets Assignment 1 branch ✔
- Issue 1: Application no longer crashes ✔
- Issue 2: Unit test properly passes ✔

# Extra Credit

- Come up with a new feature for the application
- Issue 3: Add a new feature request implemented ✔
- Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 2]
```diff
@@ -18,13 +18,19 @@ public static void Main(string[] args)
...
             }
             Console.WriteLine("You got " + GetPercentCorrect(numberCorrect, questions.Length) + " correct");
         }
 
-        public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
+        public static string GetPercentCorrect(double numberCorrectAnswers, int numberOfQuestions)
         {
-            return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+            // This method is fixed by calling the argument numberCorrectAnswers as doubles in order to properly divide them
+            double correctPercentage = (numberCorrectAnswers / numberOfQuestions) * 100;
```
Comment (Malcolm-Shepherd) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:32]: I would just leave numberCorrectAnswers as a int and cast as a double on this line. It changes the existing code less and usually you won't be getting fractions of a correct answer as an input in this situation.

Comment (mmoran314) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: Changed this to keep the original code as well as only casting one double.

File: ClassDemo/ClassDemo.Tests/Vehicle.Tests.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+﻿namespace ClassDemo.Tests;
+
+[TestClass]
+public class VehicleTests
```
Comment (mmwoodfo) [inline ClassDemo/ClassDemo.Tests/Vehicle.Tests.cs:4]: You seem to have some extra files in here from the class demo, ontop of your assignment files.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 4]
```diff
@@ -86,5 +87,25 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
                 File.AppendAllLines(filePath, lines);
             }
         }
+        [TestMethod]
+        public void GetCorrectAnswer_ReturnsExpectedString()
```
Comment (mmwoodfo) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:91]: This test method name doesn't tell me what exactly this test is doing

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 5]
```diff
@@ -86,5 +87,25 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+            //Arrange 
+            Question question = new Question();
+            question.Text = "This is the question text";
+            question.Answers = new string[3];
+            question.Answers[0] = "uno";
+            question.Answers[1] = "dos";
+            question.Answers[2] = "tres";
+            question.CorrectAnswerIndex = "2";
+            int answerIndex = Int32.Parse(question.CorrectAnswerIndex) - 1;
+            string expectedResult = question.Answers[answerIndex];
```
Comment (mmwoodfo) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:102]: I would recommend hard coding this for a test

**Pull request closed without merging**